### ISSUE TITLE
problem when typing a pacman -Syu on msys2 on Windows

### ISSUE THREAD

**Issue by auahhah** [msys2/MSYS2-packages#3944]
### Description / Steps to reproduce the issue

![‏‏لقطة الشاشة (124)](https://github.com/msys2/MSYS2-packages/assets/80679957/e4fe3200-78b1-4eb1-92c9-d285a63ad4a3)
Having a problem after installing msys2 And type pacman -Syu He delays and then shows up error Like the image I wanted to download compiler and debuuger c++ Tried several solutions but it didn't even work pacman -Ss gcc and pacman -Ss gdb Please help, thanks

### Expected behavior

I expected not to have problems 

### Actual behavior

The problem still exists and I have searched and tried several ways but it didn't work 

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

msys2 -×86_6420230718.exe

### Are you willing to submit a PR?

yes

**Comment by Biswa96** [comment]
Network issue. Make sure the programs in msys2 installation are allowed to access network in firewall.

**Comment by auahhah** [comment]
> Network issue. Make sure the programs in msys2 installation are allowed to access network in firewall.

How do I do that + I tried vpn And another network the same problem

**Comment by Biswa96** [comment]
Try to ping two or three random URLs from that error message. Can you connect to them with ping or in a web browser?

**Comment by ognevnydemon** [comment]
hm, I had this this problem when I used PdaNet program. do you use any program that's related with network connection?

**Comment by auahhah** [comment]
> hm, I had this problem when I used a PdaNet program. do you use any program that's related with the network connection?

no

**Comment by auahhah** [comment]
> @Biswa96 commented [2 hours ago](https://github.com/msys2/MSYS2-packages/issues/3944#issuecomment-1669699849):
> 
> > Can you connect to them with ping or in a web browser?
> 
> To be exact, ping uses a connectionless protocol (ICMP), so the better wording to use would be "can you _reach_ them with ..."
> 
> But instead of ping I would boot a Linux and do a TCP-based trace of route: https://support.opendns.com/hc/en-us/articles/227989007-How-to-Running-a-TCP-Traceroute
> 
> Eg. in this case:
> 
> ```shell
> $ sudo tcptraceroute mirrors.piconets.webwerks.in 443
> ```
> 
> ... or maybe use more advanced tools:
> 
> * [http-traceroute](https://github.com/watson/http-traceroute#http-traceroute)
> * [httping](https://cmdref.net/os/linux/command/httping.html)
> * [golang-httptrace](https://golang.cafe/blog/golang-httptrace-example.html#:~:text=org/p/iJ9HDabEWMu-,Expected%20result,-starting%20to%20create)
> 
> At least I would test it with cURL:
> 
> ```shell
> $ curl -vL https://mirrors.piconets.webwerks.in/
> ```
> 
> PS. @auahhah: are you connecting to web via HTTP proxy by chance?

Yes, I can connect through the browser and ping correct http Yes

**Comment by sskras** [comment]
@auahhah, so what does the cURL say in a command line above or maybe with disabled HTML output to you:
```sh
curl -vLo /dev/null https://mirrors.piconets.webwerks.in/
```
... um ?

**Comment by auahhah** [comment]
> @auahhah، فما الذي يقوله عنوان URL في سطر الأوامر أعلاه أو ربما مع تعطيل إخراج HTML لك:
> 
> ```shell
> curl -vLo /dev/null https://mirrors.piconets.webwerks.in/
> ```
> 
> ... أم؟

![‏‏لقطة الشاشة (125)](https://github.com/msys2/MSYS2-packages/assets/80679957/7b4c3276-a86d-4f90-acd7-21afd4d8f5da)

**Comment by sskras** [comment]
Cool, seems Windows version of cURL is connecting just fine.
Could you run the same command from MSYS2 and compare the outputs?

**Comment by auahhah** [comment]
> Connected to mirrors.piconets.webwerks.in (43.240.66.200) port 443 (#0)

![‏‏لقطة الشاشة (127)](https://github.com/msys2/MSYS2-packages/assets/80679957/392dee1d-c5d4-4785-8260-e4006dea0c65)
Connected to mirrors.piconets.webwerks.in (43.240.66.200) port 443 (#0)Dell@DESKTOP-DS0SPO3 UCRT64 ~
$ openssl version
OpenSSL 3.1.1 30 May 2023 (Library: OpenSSL 3.1.1 30 May 2023)

Dell@DESKTOP-DS0SPO3 UCRT64 ~
$ echo | openssl s_client -connect mirrors.piconets.webwerks.in:443 -tlsextdebug
CONNECTED(00000004)
write:errno=113
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 0 bytes and written 330 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
This TLS version forbids renegotiation.
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---

**Comment by sskras** [comment]
Thanks, I think we are getting close.

Could you please replace `-tlsextdebug` with `-msg` ?
```sh
$ echo | openssl s_client -connect mirrors.piconets.webwerks.in:443 -msg
```
This should get us the details of TLS handshake.

(No need for screenshots at this stage)

**Comment by auahhah** [comment]
> Thanks, I think we are getting close.
> 
> Could you please replace `-tlsextdebug` with `-msg` ?
> 
> ```shell
> $ echo | openssl s_client -connect mirrors.piconets.webwerks.in:443 -msg
> ```
> 
> This should get us the details of TLS handshake.
> 
> (No need for screenshots at this stage)

$ openssl s_client -connect mirrors.piconets.webwerks.in:443 -msg
CONNECTED(00000004)
>>> TLS 1.0, RecordHeader [length 0005]
    16 03 01 01 45
>>> TLS 1.3, Handshake [length 0145], ClientHello
    01 00 01 41 03 03 09 c8 7f 9c a5 c5 ac a5 24 bd
    29 a3 0c 43 3a 09 5d b4 e6 a9 98 dd 2c f6 19 58
    e9 51 5e fb f8 d0 20 73 99 18 93 37 e6 ec ff 5a
    eb 59 33 37 8d 06 6b 3f 40 57 29 69 d7 74 ca 9e
    8b ea 68 29 46 43 4a 00 3e 13 02 13 03 13 01 c0
    2c c0 30 00 9f cc a9 cc a8 cc aa c0 2b c0 2f 00
    9e c0 24 c0 28 00 6b c0 23 c0 27 00 67 c0 0a c0
    14 00 39 c0 09 c0 13 00 33 00 9d 00 9c 00 3d 00
    3c 00 35 00 2f 00 ff 01 00 00 ba 00 00 00 21 00
    1f 00 00 1c 6d 69 72 72 6f 72 73 2e 70 69 63 6f
    6e 65 74 73 2e 77 65 62 77 65 72 6b 73 2e 69 6e
    00 0b 00 04 03 00 01 02 00 0a 00 16 00 14 00 1d
    00 17 00 1e 00 19 00 18 01 00 01 01 01 02 01 03
    01 04 00 23 00 00 00 16 00 00 00 17 00 00 00 0d
    00 2a 00 28 04 03 05 03 06 03 08 07 08 08 08 09
    08 0a 08 0b 08 04 08 05 08 06 04 01 05 01 06 01
    03 03 03 01 03 02 04 02 05 02 06 02 00 2b 00 05
    04 03 04 03 03 00 2d 00 02 01 01 00 33 00 26 00
    24 00 1d 00 20 26 a5 be 62 5f 45 1d 0e a7 7f d9
    d7 a9 ef ed 19 d9 c5 58 ca 55 6e 9b 69 49 ec 9f
    88 e4 09 fb 1e
write:errno=113
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 0 bytes and written 330 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
This TLS version forbids renegotiation.
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---

**Comment by sskras** [comment]
Right, so the TCP session gets terminated just before the receiving server response:

![image](https://github.com/msys2/MSYS2-packages/assets/7887758/ec753e88-e627-4bc5-b668-c5d7230d3b81)

This happens immediately after two TLS handshake steps sending data from client to server:
```
>>> TLS 1.0, RecordHeader [length 0005]
```
```
>>> TLS 1.3, Handshake [length 0145], ClientHello
```
If I get things right, these two steps tries to send 5 + 325 = 330 bytes over network using `write()`.
Immediatelly the call responds with error = 113.

Something seemingly terminates your outgoing SSL/TLS connections.

1. Is that a private machine or maybe a work/corporate one?
2. Is your network security-restricted by default?
3. Do your browser use some proxy server?

**Comment by auahhah** [comment]
> Right, so the TCP session gets terminated just before receiving the server response:
> 
> ![image](https://user-images.githubusercontent.com/7887758/259526022-ec753e88-e627-4bc5-b668-c5d7230d3b81.png)
> 
> This happens immediately after two TLS handshake steps try to send data from client to server:
> 
> ```
> >>> TLS 1.0, RecordHeader [length 0005]
> ```
> 
> ```
> >>> TLS 1.3, Handshake [length 0145], ClientHello
> ```
> 
> If I get things right, these two steps try to send 5 + 325 = 330 bytes over network using `write()`. Immediately after that the call returns with error = 113 on your machine.
> 
> While on mine the client starts receiving TLS responses from the remote server:
> 
> ```
> <<< TLS 1.2, RecordHeader [length 0005]
> ```
> 
> ```
> <<< TLS 1.3, Handshake [length 0041], ServerHello
> ```
> 
> Something seemingly terminates your outgoing SSL/TLS connections.
> 
> 1. Is that a private machine or maybe a work/corporate one?
> 2. Is your network security-restricted by default?
> 3. Do your browser use some proxy server on that machine?

Yes, for me and the network is private, but for a proxy, no

**Comment by sskras** [comment]
Now for the last try, could you connect to `google.com`:
```
$ echo | openssl s_client -connect google.com:443 -msg
```
?

**Comment by auahhah** [comment]
> Now for the last try, could you connect to `google.com`:
> 
> ```
> $ echo | openssl s_client -connect google.com:443 -msg
> ```
> 
> ?

CONNECTED(00000004)
>>> TLS 1.0, RecordHeader [length 0005]
    16 03 01 01 33
>>> TLS 1.3, Handshake [length 0133], ClientHello
    01 00 01 2f 03 03 aa 9b 48 70 d4 d4 85 8b 2a 48
    01 05 6c ad 8b 1f 0c 94 3e e7 71 34 a7 e9 d9 05
    96 84 a7 39 29 94 20 0c 1c fe e2 6d 8d 49 36 75
    06 78 47 49 b2 b9 27 d3 aa b5 26 fc bb 1f 09 c4
    49 64 57 49 62 e7 78 00 3e 13 02 13 03 13 01 c0
    2c c0 30 00 9f cc a9 cc a8 cc aa c0 2b c0 2f 00
    9e c0 24 c0 28 00 6b c0 23 c0 27 00 67 c0 0a c0
    14 00 39 c0 09 c0 13 00 33 00 9d 00 9c 00 3d 00
    3c 00 35 00 2f 00 ff 01 00 00 a8 00 00 00 0f 00
    0d 00 00 0a 67 6f 6f 67 6c 65 2e 63 6f 6d 00 0b
    00 04 03 00 01 02 00 0a 00 16 00 14 00 1d 00 17
    00 1e 00 19 00 18 01 00 01 01 01 02 01 03 01 04
    00 23 00 00 00 16 00 00 00 17 00 00 00 0d 00 2a
    00 28 04 03 05 03 06 03 08 07 08 08 08 09 08 0a
    08 0b 08 04 08 05 08 06 04 01 05 01 06 01 03 03
    03 01 03 02 04 02 05 02 06 02 00 2b 00 05 04 03
    04 03 03 00 2d 00 02 01 01 00 33 00 26 00 24 00
    1d 00 20 b2 cf 8f c7 2b fd e8 40 2b 52 16 d8 1d
    79 68 7d ee 9f c6 43 67 27 b1 a9 35 61 8e 23 3e
    0c 05 6f
write:errno=113
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 0 bytes and written 312 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
This TLS version forbids renegotiation.
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---

**Comment by sskras** [comment]
@auahhah, first goes my suggestion about posting.

You seem to not properly quote output that you paste from MinTTY window (so @Biswa96 has to fix it for the readers).
This is the proper syntax:
````
```
CONNECTED(00000004)
>>> TLS 1.0, RecordHeader [length 0005]
    16 03 01 01 33
>>> TLS 1.3, Handshake [length 0133], ClientHello
    01 00 01 2f 03 03 aa 9b 48 70 d4 d4 85 8b 2a 48
    01 05 6c ad 8b 1f 0c 94 3e e7 71 34 a7 e9 d9 05
    96 84 a7 39 29 94 20 0c 1c fe e2 6d 8d 49 36 75
    06 78 47 49 b2 b9 27 d3 aa b5 26 fc bb 1f 09 c4
    49 64 57 49 62 e7 78 00 3e 13 02 13 03 13 01 c0
    2c c0 30 00 9f cc a9 cc a8 cc aa c0 2b c0 2f 00
    9e c0 24 c0 28 00 6b c0 23 c0 27 00 67 c0 0a c0
    14 00 39 c0 09 c0 13 00 33 00 9d 00 9c 00 3d 00
    3c 00 35 00 2f 00 ff 01 00 00 a8 00 00 00 0f 00
    0d 00 00 0a 67 6f 6f 67 6c 65 2e 63 6f 6d 00 0b
    00 04 03 00 01 02 00 0a 00 16 00 14 00 1d 00 17
    00 1e 00 19 00 18 01 00 01 01 01 02 01 03 01 04
    00 23 00 00 00 16 00 00 00 17 00 00 00 0d 00 2a
    00 28 04 03 05 03 06 03 08 07 08 08 08 09 08 0a
    08 0b 08 04 08 05 08 06 04 01 05 01 06 01 03 03
    03 01 03 02 04 02 05 02 06 02 00 2b 00 05 04 03
    04 03 03 00 2d 00 02 01 01 00 33 00 26 00 24 00
    1d 00 20 b2 cf 8f c7 2b fd e8 40 2b 52 16 d8 1d
    79 68 7d ee 9f c6 43 67 27 b1 a9 35 61 8e 23 3e
    0c 05 6f
write:errno=113
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 0 bytes and written 312 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
This TLS version forbids renegotiation.
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---

```
````
In other words, to stay clean in future, you should insert ```` ``` ```` lines at the start and at the end of the pasted text.

**Comment by sskras** [comment]
Regarding your issue with MSYS2.

I see same outcome. MSYS2 installed binaries are unable to start establishing HTTPS session even to `google.com`.
Comparing to your output for `mirrors.piconets.webwerks.in`:

![image](https://github.com/msys2/MSYS2-packages/assets/7887758/c3d0f89b-2210-45b5-b182-38ef91492ef9)

... the first TCP write/send to Google gets aborted too.  Weird.
Now I would boot this machine into safer (or different) OS environment like:

1. Windows in [Safe Mode with Networking](https://www.bleepingcomputer.com/tutorials/how-to-start-windows-10-in-safe-mode-with-networking/) + MSYS2;
2. Live Media on any [Linux distribution appropriate](https://www.fosslinux.com/274/how-to-create-linux-mint-live-usb-drive-on-windows.htm#attachment_49041:~:text=Creating%20Linux%20Mint%20OS%20live%20USB%20on%20Windows) for your machine.

... and would retry cURL tests from here.

In case cURL request succeeds at least in one of these environments, it would mean that some software from your Windows installation interferes with the SSL connections made by MSYS2 binaries.

If that's not the case, then termination must be happening somewhere along the route to a server.
I have no ideas yet on how to identify exactly which hop does that.

**Comment by sskras** [comment]
@auahhah, just in case, let's check plain-HTTP connection. For me it works too:
```
$ curl -svo /dev/null http://mirrors.piconets.webwerks.in/
*   Trying 43.240.66.200:80...
* Connected to mirrors.piconets.webwerks.in (43.240.66.200) port 80 (#0)
> GET / HTTP/1.1
> Host: mirrors.piconets.webwerks.in
> User-Agent: curl/8.0.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: nginx
< Date: Thu, 10 Aug 2023 08:16:10 GMT
< Content-Type: text/html
< Transfer-Encoding: chunked
< Connection: keep-alive
< Expires: Thu, 10 Aug 2023 14:16:10 GMT
< Cache-Control: max-age=21600
< Cache-Control: public
<
{ [2383 bytes data]
* Connection #0 to host mirrors.piconets.webwerks.in left intact
```

**Comment by auahhah** [comment]
> Regarding your issue with MSYS2.
> 
> I see same outcome. MSYS2 installed binaries are unable to start establishing HTTPS session even to `google.com`. Comparing to your output for `mirrors.piconets.webwerks.in`:
> 
> ![image](https://user-images.githubusercontent.com/7887758/259669252-c3d0f89b-2210-45b5-b182-38ef91492ef9.png)
> 
> ... the first TCP write/send to Google gets aborted too. Weird. Now I would boot this machine into safer (or different) OS environment like:
> 
> 1. Windows in [Safe Mode with Networking](https://www.bleepingcomputer.com/tutorials/how-to-start-windows-10-in-safe-mode-with-networking/) + MSYS2;
> 2. Live Media on any [Linux distribution appropriate](https://www.fosslinux.com/274/how-to-create-linux-mint-live-usb-drive-on-windows.htm#attachment_49041:~:text=Creating%20Linux%20Mint%20OS%20live%20USB%20on%20Windows) for your machine.
> 
> ... and would retry cURL tests from here.
> 
> In case cURL request succeeds at least in one of these environments, it would mean that some software from your Windows installation interferes with the SSL connections made by MSYS2 binaries.
> 
> If that's not the case, then termination must be happening somewhere along the route to a server. I have no ideas yet on how to identify exactly which hop does that.

I have kali linux on vmware workstation 17 pro

**Comment by sskras** [comment]
> I have kali linux on vmware workstation 17 pro

That would be a VM, different machine I guess.

I suggest to boot Kali Linux on the same machine. Eg:
https://turbofuture.com/computers/How-to-Run-Backtrack-5-From-a-Bootable-USB-Drive#booting-kali-linux

One change at a time makes picture more clear. :)

**Comment by auahhah** [comment]
> > I have kali linux on vmware workstation 17 pro
> 
> That would be a VM, different machine I guess.
> 
> I suggest to boot Kali Linux on the same machine. Eg: https://turbofuture.com/computers/How-to-Run-Backtrack-5-From-a-Bootable-USB-Drive#booting-kali-linux
> 
> One change at a time makes picture more clear. :)
> 
> EDIT: Booting your Windows in **Safe Mode with Networking** (as mentioned before) would be the most minimal change, and the fastest one.
> 
> EDIT2: It would be still interesting to see if **plain-HTTP** (non-SSL/TLS) **connections** are also terminated from your MSYS2 ([as asked above](#issuecomment-1672767798)).

Linux Mint is installed, what's next?

**Comment by sskras** [comment]
> Linux Mint is installed, what's next?

Already wrote:
https://github.com/msys2/MSYS2-packages/issues/3944#issuecomment-1672767798

PS. Installing was unnecessary. Booting from Mint Live USB/media would have been enough. :)

**Comment by auahhah** [comment]
> > Linux Mint is installed, what's next?
> 
> Already mentioned: [#3944 (comment)](https://github.com/msys2/MSYS2-packages/issues/3944#issuecomment-1670279877) (fixed link to the wrong comment)
> 
> PS. Installing was unnecessary. Booting from Mint Live USB/media would have been enough. :) Fingers crossed.

I installed a sub-copy of Windows 11 for the original Windows. It worked perfectly on the sub. I think the problem was from the original Windows.

**Comment by sskras** [comment]
Nice to hear. Should the issue be closed?

**Comment by auahhah** [comment]
> Nice to hear. Should the issue be closed?

Yes, thank you for everything. I appreciate you
